src/writer.js

```diff
@@ -7,32 +7,55 @@ const encode = require('./encode')
 const tracerVersion = require('../lib/version')
 
 class Writer {
-  constructor (url, size) {
+  constructor (url, size, processors) {
     this._queue = []
     this._url = url
     this._size = size
+    this._processors = processors
   }
 
   get length () {
     return this._queue.length
   }
 
+  // A trace is considered filtered out if it is undefined, or if it contains no spans
+  isTraceFilteredOut (trace) {
+    return trace === undefined || trace === []
+  }
+
+  applyProcessors (trace) {
+    return this._processors.reduce((currentTrace, processor) => {
+      // If the trace has been filtered out by a previous processor, we don't need
+      // to bother calling the rest
+      if (!this.isTraceFilteredOut(currentTrace)) {
+        return processor(currentTrace)
+      }
+    }, trace)
+  }
+
   append (span) {
     const trace = span.context().trace
 
     if (trace.started.length === trace.finished.length) {
-      const formattedTrace = trace.finished.map(format)
+      log.debug(() => `Formatting original trace: ${JSON.stringify(trace.finished)}`)
 
-      log.debug(() => `Encoding trace: ${JSON.stringify(formattedTrace)}`)
+      const formattedTrace = trace.finished.map(format)
 
-      const buffer = encode(formattedTrace)
+      log.debug(() => `Running formatted trace through any configured processors: ${JSON.stringify(formattedTrace)}`)
```

Comment:
I know it's my fault for doing this initially 😅but it would probably make more sense to put these log entries in the formatter/encoder/processor directly.